samples/csharp_dotnetcore/07.using-adaptive-cards/Startup.cs

```diff
@@ -20,8 +32,23 @@ public void ConfigureServices(IServiceCollection services)
             // Create the Bot Framework Adapter with error handling enabled.
             services.AddSingleton<IBotFrameworkHttpAdapter, AdapterWithErrorHandler>();
 
-            // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
-            services.AddTransient<IBot, AdaptiveCardsBot>();
+            // Create the bot. the ASP Controller is expecting an IBot.
+            services.AddSingleton<IBot, DialogBot<RootDialog>>();
+
+            // The adaptive dialog that will be run by the bot.
+            services.AddSingleton<RootDialog>();
+
+            // Create the credential provider to be used with the Bot Framework Adapter.
+            services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
```

Comment:
This can be removed, and AdapterWithErrorHandler updated as follows:

```cs
        public AdapterWithErrorHandler(ILogger<BotFrameworkHttpAdapter> logger, IStorage storage,
            UserState userState, ConversationState conversationState, IConfiguration configuration)
            : base(configuration, logger)
```